service/src/main/java/bio/terra/pipelines/stairway/WriteJobToDbStep.java

```diff
@@ -0,0 +1,59 @@
+package bio.terra.pipelines.stairway;
+
+import bio.terra.pipelines.common.utils.FlightUtils;
+import bio.terra.pipelines.dependencies.stairway.StairwayJobMapKeys;
+import bio.terra.pipelines.service.JobsService;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import java.time.Instant;
+import java.util.Objects;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.retry.RetryException;
+
+public class WriteJobToDbStep implements Step {
+  private final JobsService jobsService;
+  private final Logger logger = LoggerFactory.getLogger(WriteJobToDbStep.class);
+
+  public WriteJobToDbStep(JobsService jobsService) {
+    this.jobsService = jobsService;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext)
+      throws InterruptedException, RetryException {
+    var inputParameters = flightContext.getInputParameters();
+    FlightUtils.validateRequiredEntries(
+        inputParameters,
+        CreateJobFlightMapKeys.PIPELINE_ID,
+        CreateJobFlightMapKeys.PIPELINE_VERSION,
+        CreateJobFlightMapKeys.SUBMITTING_USER_ID);
+
+    var workingMap = flightContext.getWorkingMap();
+    FlightUtils.validateRequiredEntries(workingMap, CreateJobFlightMapKeys.STATUS);
+
+    UUID writtenJobUUID =
+        jobsService.writeJobToDb(
+            UUID.fromString(flightContext.getFlightId()),
```

Comment:
hmm generally you dont want to have to keep going from strings to UUIDs and vice versa in different parts of the code.  Is there any way to streamline this so its always a UUID wherever the value is?

Reply:
I made some changes, but this particular one I'm having trouble getting around, since FlightContext is a Stairway class, and .getFlight() returns a String. I agree that it's gross to keep going back and forth... open to suggestions here!

Reply:
i think the changes you've made are alot better.  All of the transitiions between types are at teh stairway interaction level where I think it makes the most sense for it to be